src/ln/msgs.rs

```diff
@@ -562,8 +564,14 @@ pub trait RoutingMessageHandler : Send + Sync {
 	fn handle_channel_update(&self, msg: &ChannelUpdate) -> Result<bool, HandleError>;
 	/// Handle some updates to the route graph that we learned due to an outbound failed payment.
 	fn handle_htlc_fail_channel_update(&self, update: &HTLCFailChannelUpdate);
+	///This returns a vec of vec's, one for ChannelAnnouncements, one for ChannelUpdates and one for NodeAnnouncements. It also returns the id of the last channel passed through
+	fn get_next_announcements(&self, starting_point: u64, batch_amount: u8)->(Vec<ChannelAnnouncement>, Vec<ChannelUpdate>, Vec<NodeAnnouncement>, u64);
```

Comment:
This just feels like needless overhead - cant we do a (ChannelAnnouncement, NodeAnnouncement, ChannelUpdate, ChannelUpdate)?

Reply:
are you suggesting we do single batches?

Reply:
The reason I split them up into separate vec, and not a vec<tuple> is to avoid having to check if we have one for each. We should have but this way we dont have to check it and its up to the remote peer to check them as we send them separately as well.  The remote peer is going to check them anyway.

Reply:
We can't send channel_updates for channels we haven't already channel_announcement'd, nor can we send node_announcements for channels we haven't channel_announcement'd, that's a protocol violation. So we can't rely on the remote peer to check it, we have to make sure what we're sending is sane.